tidy3d/components/heat/simulation.py

```diff
@@ -116,24 +111,66 @@ def check_unsupported_geometries(cls, val):
+        a ``SolidSpec`` medium.
+        """
 
-            if isinstance(structure.geometry, GeometryGroup):
-                geometries = structure.geometry.geometries
+        # list of structures including background as a Box()
+        structure_bg = Structure(
+            geometry=Box(
+                size=values.get("size"),
+                center=values.get("center"),
+            ),
+            medium=values.get("medium"),
+        )
+
+        structures = values.get("structures") or []
+        total_structures = [structure_bg] + list(structures)
+
+        failed_obj_inds = []
+        for ind, obj in enumerate(objs):
+            if obj.size.count(0.0) == 1:
+                # for planar objects we could do a rigorous check
+                medium_set = Scene.intersecting_media(obj, total_structures)
+                crosses_solid = any(
+                    isinstance(medium.heat_spec, SolidSpec) for medium in medium_set
+                )
             else:
-                geometries = [structure.geometry]
-            for geom in geometries:
-                if isinstance(geom, (GeometryGroup)):
-                    raise SetupError(
-                        "'HeatSimulation' does not currently support recursive 'GeometryGroup's ('structures[{ind}]')."
-                    )
-                if not isinstance(geom, HeatSingleGeometryType):
-                    geom_names = [f"'{cl.__name__}'" for cl in HeatSingleGeometryType]
-                    raise SetupError(
-                        "'HeatSimulation' does not currently support geometries of type "
-                        f"'{geom.type}'  ('structures[{ind}]'). Allowed geometries are "
-                        f"{', '.join(geom_names)}, "
-                        "and non-recursive 'GeometryGroup'."
-                    )
+                # approximate check for volumetric objects based on bounding boxes
+                # thus, it could still miss a case when there is no data inside the monitor
+                crosses_solid = any(
+                    obj.intersects(structure.geometry)
+                    for structure in total_structures
+                    if isinstance(structure.medium.heat_spec, SolidSpec)
+                )
+
+            if not crosses_solid:
+                failed_obj_inds.append(ind)
+
+        return failed_obj_inds
+
+    @pd.validator("monitors", always=True)
+    @skip_if_fields_missing(["medium", "center", "size", "structures"])
+    def _monitors_cross_solids(cls, val, values):
+        """Error if monitors does not cross any solid medium."""
+
+        if val is None:
+            return val
+
+        failed_mnt_inds = cls._check_cross_solids(val, values)
+
+        if len(failed_mnt_inds) > 0:
+            monitor_names = ", ".join(f"'{val[ind].name}'" for ind in failed_mnt_inds)
+            raise SetupError(
+                f"Monitor(s) {monitor_names} do(es) not cross any solid materials ('heat_spec=SolidSpec(...)'). "
```

Comment:
really minor, but I think it would be fine to just do something like
```py
monitor_names = [f"'{val[ind].name}'" for ind in failed_mnt_inds)]
raise SetupError(
                f"Monitors {monitor_names} do not cross any solid materials ('heat_spec=SolidSpec(...)'). "
)
```